### PR TITLE
implement general mark join

### PR DESCRIPTION
**Pull request by JackTan25**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

**Comment by JackTan25** [comment]
I don't know whose pr result in a conflict. Before I fix the conflict, this pr can pass all tests. But when I 'git pull upstream main' and fix the conflicts. It's failed. I don't know why.

**Review by fengttt — CHANGES_REQUESTED** [review]
Can you please add a bvt test, a few tests, join on a few different types int, decimal, string, etc that trigger this mark join?

File: pkg/sql/compile/compile.go [hunk 1]
```diff
@@ -693,9 +693,9 @@ func (c *Compile) compileJoin(n, right *plan.Node, ss []*Scope, children []*Scop
 		for i := range rs {
 			if isEq && len(conds) == 1 {
```
Comment (nnsgmsone) [inline pkg/sql/compile/compile.go:694]: Please change it to:
```
if isEq {

**Review by nnsgmsone — CHANGES_REQUESTED** [review]

File: pkg/sql/compile/compile.go (hunk 1 above)
Comment (nnsgmsone) [reply pkg/sql/compile/compile.go:694]: mark join not anti join

**Comment by nnsgmsone** [comment]
> Can you please add a bvt test, a few tests, join on a few different types int, decimal, string, etc that trigger this mark join?

bvt may not be necessary, as some of the subqueries in the original bvt I had handled with nested loop anti. These should now be handled with a hash mark join

**Comment by JackTan25** [comment]
> > Can you please add a bvt test, a few tests, join on a few different types int, decimal, string, etc that trigger this mark join?
> 
> bvt may not be necessary, as some of the subqueries in the original bvt I had handled with nested loop anti. These should now be handled with a hash mark join

ok

**Comment by fengttt** [comment]
> > Can you please add a bvt test, a few tests, join on a few different types int, decimal, string, etc that trigger this mark join?
> 
> bvt may not be necessary, as some of the subqueries in the original bvt I had handled with nested loop anti. These should now be handled with a hash mark join

Best if you have a bvt test case, call it markjoin.sql/test, and have some queries that explicitly, surely trigger that code.   You may want to run explain of the query and check the output that it is indeed a mark join.

**Comment by nnsgmsone** [comment]
> Best if you have a bvt test case, call it markjoin.sql/test, and have some queries that explicitly, surely trigger that code. You may want to run explain of the query and check the output that it is indeed a mark join.

Makes sense

**Comment by JackTan25** [comment]
wait，I need Jensen to modify the join_test, so please ignore the result of the CI

**Pull request closed without merging**